server/src/main/webapp/WEB-INF/rails/webpack/views/components/behavior_prompt/behavior_prompt.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {bind} from "classnames/bind";
+import {mixins as s} from "helpers/string-plus";
+import {MithrilComponent} from "jsx/mithril-component";
+import _ from "lodash";
+import m from "mithril";
+import Stream from "mithril/stream";
+import * as style from "./behavior_prompt.scss";
+
+const classnames = bind(style);
+type Styles = typeof style;
+
+interface Attrs {
+  promptText: string;
+  key: string;
+  query: string;
+  direction: Direction;
+  size?: Size;
+  position?: Position;
+}
+
+export const STORAGE_KEY_PREFIX = "gocd.behaviorPrompt";
+
+export class BehaviorPrompt extends MithrilComponent<Attrs> {
+  show: Stream<boolean> = Stream();
+
+  oninit(vnode: m.Vnode<Attrs>) {
+    const storageKey = STORAGE_KEY_PREFIX.concat(".", vnode.attrs.key);
+    if (!localStorage.getItem(storageKey)) {
```

Comment:
This works, but would it be more clear to explicitly check that it's not null or false?